src/System.IO.FileSystem/src/System/IO/DirectoryInfo.Unix.cs

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.IO
+{
+    partial class DirectoryInfo
+    {
+        private static bool SpecialCaseToCurrentDirectory(string path)
```

Comment:
I'm going to wait to merge these changes until #2267 is merged.  Then I'll probably move this helper to PathHelpers.Windows/Unix.cs rather than making DirectoryInfo a partial class.